File: src/commands/env.ts
import Command from '../command';
import { Trace } from '@aiteq/trace';

@Trace()
export default class Env extends Command {
  constructor() {
    super('env');
  }

  public async execute() {
    const args = ['exec', '-ti', this.id, 'env'];
    return this.spawn('docker', args);
  }
}

File: src/commands/env.rs
use crate::utils::{command, exec, prompt};

/// Ease of use shortcut to log the environment variables of a running container
pub fn run(sub_matches: &clap::ArgMatches) {
    let mut args = vec!["exec".to_string(), "-ti".to_string()];
    let choice = if let Some((query, _)) = sub_matches.subcommand() {
        command::handle_fuzzy_search(query, "env")
    } else {
        prompt::prompt("Select a container from the list", "env")
    };

    args.extend(vec![choice, "env".to_string()]);

    exec::run_and_exit(&args);
}
